viewer/packages/camera-manager/src/CameraManager.ts

```diff
@@ -0,0 +1,590 @@
+
+    // 3. Handle when camera is inside the model by adjusting the near value
+    const diagonal = combinedBbox.min.distanceTo(combinedBbox.max);
+    if (combinedBbox.containsPoint(cameraPosition)) {
+      near = Math.min(0.1, far / 1000.0);
+    }
+
+    // Apply
+    if (this.automaticNearFarPlane) {
+      camera.near = near;
+      camera.far = far;
+      camera.updateProjectionMatrix();
+    }
+    if (this.automaticControlsSensitivity) {
+      // The minDistance of the camera controller determines at which distance
+      // we will stop when zooming with mouse wheel.
+      // This is also used to determine the speed of the camera when flying with ASDW.
+      // We want to either let it be controlled by the near plane if we are far away,
+      // but no more than a fraction of the bounding box of the system if inside
+      this._controls.minDistance = Math.min(Math.max(diagonal * 0.02, 0.1 * near), CameraManager.DefaultMinDistance);
+    }
+  }
+
+  updateCameraControlsState(deltaTime: number): void {
+    this._controls.update(deltaTime);
+  }
+
+  dispose(): void {
+    this.isDisposed = true;
+    this._controls.dispose();
+    this.teardownControls();
+  }
+
+  private calculateAnimationStartTarget(newTarget: THREE.Vector3): THREE.Vector3 {
+    const { _raycaster, _camera } = this;
+    _raycaster.setFromCamera(new THREE.Vector2(), _camera);
+    const distanceToTarget = newTarget.distanceTo(_camera.position);
+    const scaledDirection = _raycaster.ray.direction.clone().multiplyScalar(distanceToTarget);
+
+    return _raycaster.ray.origin.clone().add(scaledDirection);
+  }
+
+  private createTweenAnimation(
+    from: any,
+    to: any,
+    duration: number
+  ): { tween: TWEEN.Tween; stopTween: (event: Event) => void } {
+    const animation = new TWEEN.Tween(from);
+    const stopTween = (event: Event) => {
+      if (this.isDisposed) {
+        document.removeEventListener('keydown', stopTween);
+        animation.stop();
+        return;
+      }
+
+      if (event.type !== 'keydown' || this._controls.enableKeyboardNavigation) {
+        animation.stop();
+        this._domElement.removeEventListener('pointerdown', stopTween);
+        this._domElement.removeEventListener('wheel', stopTween);
+        document.removeEventListener('keydown', stopTween);
+      }
+    };
+
+    this._domElement.addEventListener('pointerdown', stopTween);
+    this._domElement.addEventListener('wheel', stopTween);
+    document.addEventListener('keydown', stopTween);
+
+    const tween = animation.to(to, duration).easing((x: number) => TWEEN.Easing.Circular.Out(x));
+
+    return { tween, stopTween };
+  }
+
+  private calculateCameraFar(
+    near: number,
+    cameraPosition: THREE.Vector3,
+    cameraDirection: THREE.Vector3,
+    corners: Array<THREE.Vector3>
+  ): number {
+    const { nearPlane, nearPlaneCoplanarPoint } = this._updateNearAndFarPlaneBuffers;
```

Comment:
It's a bit sketchy to fetch vars from buffers from another "vars buffer". Move the stuff you need to a new buffer 
`_calculateCameraFarBuffers` or something so it's clear where the vars are used.